homeworks/gaalistvantamas/7_extract_transform/api_get_data.py

```diff
@@ -0,0 +1,53 @@
+"""
+Author: Gaál István Tamás
+Task: Homework-7
+"""
+import requests
+import pandas as pd
+import json
+
+url = "https://api.coingecko.com/api/v3/coins/markets"
+
+params = {"vs_currency": "usd",
+          "order=": "market_cap_desc",
+          "per_page": 250,}
+
+with open("homeworks/gaalistvantamas/7_extract_transform/coins.json", "w") as file:
+    cripto_coins = requests.get(url=url, params=params).json()
+    file.write(json.dumps(cripto_coins))
+
+df = pd.read_json("homeworks/gaalistvantamas/7_extract_transform/coins.json")
+#with pd.option_context('display.max_rows', None, 'display.max_columns', None):
+#    print(df)
+# 1. 
+df_Nan = pd.DataFrame(df).count()
+print("\n1...........")
+print(250-df_Nan)
```

Comment:
ez a megoldás most, hogy 250 rekordunk van működik, de amint ez megváltozik, rossz eredményeket kapunk. Próbáld így:

`df.isnull().sum()`

Reply:
Szia, ez sokkal egyszerűbb. :) Többnyire kerülni szoktam az ilyen konstansokat, de most valamiért bent maradt. Az jutott  még az eszembe akkor, amikor ezt megemlítetted, hogy használhattam volna ezt is:  params["per_page"]-df_Nan . Mert ezzel is leköveti a darabszám változást. :)

Reply:
A per_page ugye azt adja meg, hogy egy oldalon hány rekordot ad vissza. Ha több oldalnyi adatot húzol (pl 500-at két oldalon) akkor ez az érték továbbra is 250 marad és szintén nem jó eredmény születik. A legtutibb a df.isnull().sum().